### PR TITLE
Get rid of mocha tests in actionpack - part 1

### PR DESCRIPTION
**Pull request by rodzyn**
The idea is to gradually introducing minitest stubs/mocks and at the end get rid of mocha dependency.
To not making too big PRs that are hard to review, I thought I split this task into series of 2 (maybe 3) PRs.
I also removed some unused requires/stubs.

### CODE REVIEW

File: actionpack/test/dispatch/cookies_test.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
 require 'active_support/key_generator'
 require 'active_support/message_verifier'
 
+
```
Comment (kaspth) [inline actionpack/test/dispatch/cookies_test.rb]: :scissors: this line

File: actionpack/test/dispatch/cookies_test.rb [hunk 2]
```diff
@@ -321,10 +322,11 @@ def test_setting_cookie_with_secure
   end
 
   def test_setting_cookie_with_secure_when_always_write_cookie_is_true
-    ActionDispatch::Cookies::CookieJar.any_instance.stubs(:always_write_cookie).returns(true)
+    @request.cookie_jar.always_write_cookie = true
     get :authenticate_with_secure
     assert_cookie_header "user_name=david; path=/; secure"
     assert_equal({"user_name" => "david"}, @response.cookies)
+    @request.cookie_jar.always_write_cookie = false
```
Comment (kaspth) [inline actionpack/test/dispatch/cookies_test.rb]: There's a standard pattern of capturing the old value and then setting the variable back to that. Like this:

```ruby
def test_setting_cookie_with_secure_when_always_write_cookie_is_true
  old_cookie, @request.cookie_jar.always_write_cookie = @request.cookie_jar.always_write_cookie, true

  get :authenticate_with_secure
  # ...
ensure
  @request.cookie_jar.always_write_cookie = old_cookie
end
```

File: actionpack/test/dispatch/exception_wrapper_test.rb [hunk 3]
```diff
@@ -17,8 +17,6 @@ def backtrace
     end
 
     setup do
-      Rails.stubs(:root).returns(Pathname.new('.'))
```
Comment (kaspth) [inline actionpack/test/dispatch/exception_wrapper_test.rb]: Why don't we need this anymore?

File: actionpack/test/dispatch/test_request_test.rb [hunk 4]
```diff
@@ -54,9 +54,7 @@ class TestRequestTest < ActiveSupport::TestCase
   end
 
   test "does not complain when Rails.application is nil" do
-    Rails.stubs(:application).returns(nil)
```
Comment (kaspth) [inline actionpack/test/dispatch/test_request_test.rb]: Why don't we need this anymore?

The test name feels odd without this stub.

File: actionpack/test/controller/request_forgery_protection_test.rb [hunk 5]
```diff
@@ -529,11 +542,11 @@ def teardown
 
   def test_should_not_warn_if_form_authenticity_param_matches_form_authenticity_token
     ActionController::Base.logger = @logger
-    @controller.stubs(:valid_authenticity_token?).returns(:true)
-
     begin
-      post :index, params: { custom_token_name: 'foobar' }
-      assert_equal 0, @logger.logged(:warn).size
+      @controller.stub :valid_authenticity_token?, :true do
```
Comment (kaspth) [inline actionpack/test/controller/request_forgery_protection_test.rb]: What does `:true` mean here? I haven't seen that before.

File: actionpack/test/controller/request_forgery_protection_test.rb [hunk 6]
```diff
@@ -1,5 +1,4 @@
 require 'abstract_unit'
-require 'digest/sha1'
```
Comment (kaspth) [inline actionpack/test/controller/request_forgery_protection_test.rb]: Why don't we need this anymore?

File: actionpack/test/controller/render_test.rb [hunk 7]
```diff
@@ -1,6 +1,5 @@
 require 'abstract_unit'
 require 'controller/fake_models'
-require 'pathname'
```
Comment (kaspth) [inline actionpack/test/controller/render_test.rb]: Why was this removed?

**Comment by kaspth** [comment]
Thanks for working on this :+1:

File: actionpack/test/dispatch/cookies_test.rb (hunk 2 above)
Comment (rodzyn) [inline actionpack/test/dispatch/cookies_test.rb]: @kaspth Nice! Thanks for a tip :)

File: actionpack/test/dispatch/test_request_test.rb (hunk 4 above)
Comment (rodzyn) [inline actionpack/test/dispatch/test_request_test.rb]: @kaspth (the same as one example up there) During testing I found out that it's working without this one stubbed. Rails.application is nil by default in this case - without stubbing it. So, better stub it or change the message?

File: actionpack/test/controller/request_forgery_protection_test.rb (hunk 5 above)
Comment (rodzyn) [inline actionpack/test/controller/request_forgery_protection_test.rb]: @kaspth Take a look few lines up in the diff, I just moved this line inside `begin; end;` block.

File: actionpack/test/controller/request_forgery_protection_test.rb (hunk 6 above)
Comment (rodzyn) [inline actionpack/test/controller/request_forgery_protection_test.rb]: @kaspth This requirement looks like some old stuff, I haven't found it used in this test currently (also it's green without it)

File: actionpack/test/controller/render_test.rb (hunk 7 above)
Comment (rodzyn) [inline actionpack/test/controller/render_test.rb]: @kaspth git grep Pathname -- actionpack/test/controller/render_test.rb returns nothing (and tests are green without it)

File: actionpack/test/dispatch/cookies_test.rb (hunk 2 above)
Comment (rodzyn) [inline actionpack/test/dispatch/cookies_test.rb]: @kaspth Changed

File: actionpack/test/dispatch/cookies_test.rb (hunk 1 above)
Comment (rodzyn) [inline actionpack/test/dispatch/cookies_test.rb]: fixed

File: actionpack/test/dispatch/test_request_test.rb (hunk 4 above)
Comment (kaspth) [inline actionpack/test/dispatch/test_request_test.rb]: Yeah lets change the message here (and in the other test you're referring to - if that's appropriate).

File: actionpack/test/dispatch/cookies_test.rb [hunk 8]
```diff
@@ -321,10 +321,12 @@ def test_setting_cookie_with_secure
   end
 
   def test_setting_cookie_with_secure_when_always_write_cookie_is_true
-    ActionDispatch::Cookies::CookieJar.any_instance.stubs(:always_write_cookie).returns(true)
+    old__cookie, @request.cookie_jar.always_write_cookie = @request.cookie_jar.always_write_cookie, true
```
Comment (kaspth) [inline actionpack/test/dispatch/cookies_test.rb]: Lets use a single underbar for the old cookie variable.

File: actionpack/test/dispatch/test_request_test.rb (hunk 4 above)
Comment (rodzyn) [inline actionpack/test/dispatch/test_request_test.rb]: done

**Comment by kaspth** [comment]
Thanks :+1:

**Comment by rodzyn** [comment]
@kaspth Thanks for a quick feedback!
